app/src/main/java/com/github/wnder/picture/UploadInfo.java

```diff
@@ -0,0 +1,142 @@
+package com.github.wnder.picture;
+
+import android.location.Location;
+import android.net.Uri;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+
+/**
+ * This allows to create, store, load and delete objects with all the information about an
+ * a picture that is currently uploading (in order to restart the upload)
+ */
+public class UploadInfo {
+
+    private static String USER_NAME = "user_name";
+    private static String LATITUDE = "latitude";
+    private static String LONGITUDE = "longitude";
+    private static String FILE_URI = "file_uri";
+
+    String userName;
+    Location location;
+    Uri pictureUri;
+
+    /**
+     * Store a file with the information for an upload
+     * @param file the file store to
+     * @param uploadInfo the infos to store
+     * @throws JSONException
+     * @throws IOException
+     */
+    public static void storeUploadInfo(File file, UploadInfo uploadInfo) throws JSONException, IOException {
+        //Create json
+        JSONObject json = new JSONObject();
+        json.put(USER_NAME, uploadInfo.userName);
+        json.put(LATITUDE, uploadInfo.location.getLatitude());
+        json.put(LONGITUDE, uploadInfo.location.getLongitude());
+        json.put(FILE_URI, uploadInfo.pictureUri.toString());
+
+        // Store in file
+        FileOutputStream fos = new FileOutputStream(file);
+        fos.write(json.toString().getBytes());
+        fos.close();
+    }
+
+    /**
+     * Get the upload information from a file
+     * @param file the file to load from
+     * @return a UploadInfo object, null if it fails at any point
+     */
+    public static UploadInfo loadUploadInfo(File file) {
+        if(file.exists()) {
+            StringBuilder stringBuilder = new StringBuilder();
+            // Try reading the file
+            try {
+                FileInputStream fis = new FileInputStream(file);
+                InputStreamReader inputStreamReader =
+                        new InputStreamReader(fis, StandardCharsets.UTF_8);
+                try (BufferedReader reader = new BufferedReader(inputStreamReader)) {
+                    String line = reader.readLine();
+                    while (line != null) {
+                        stringBuilder.append(line);
+                        line = reader.readLine();
+                    }
+                    fis.close();
```

Comment:
If you can make a function for that try catche you should be able to remove 2 codeclimate issues I believe

Reply:
It only removed one :(

Reply:
sad